.github/workflows/lambda-runner-binaries-syncer.yml

```diff
@@ -0,0 +1,31 @@
+name: Lambda Runner Binaries Syncer
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - .github/workflows/lambda-runner-binaries-syncer.yml
+      - "terraform-aws-github-runner/modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/**"
+
+jobs:
+  build:
+    name: "Run tests for runner binaries syncer lambda"
+    runs-on: ubuntu-latest
+    container: node:20
+    defaults:
+      run:
+        working-directory: terraform-aws-github-runner/modules/runner-binaries-syncer/lambdas/runner-binaries-syncer
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: yarn install
```

Comment:
huh, this command is failing, the cause isn't completely clear though. Does it run locally? 
https://github.com/pytorch/test-infra/actions/runs/10637289034/job/29747934643?pr=5615

Reply:
Weird, yes it worked locally since I was able to build and deploy this to Canary. I'll investigate.

Reply:
@ZainRizvi can you authorize to retry the job? I pushed an update that regenerates the yarn.lock file. I think it should go through now.

Reply:
Didn't work unfortunately. You might be able to iterate on this PR faster if you fork it to a private repo while testing the changes

Reply:
Darn. I guess likely this code hasn't been tested in awhile so it's tests are broken and we have to resolve it if we want to put it back in. I'll see if I can resolve the tests here and create a separate PR for that.

Reply:
I decided to comment out the `yarn test` since I found there's many broken tests with the runner-binaries-syncer at the moment and its been broken for at least 2 years now likely because the tests for this function haven't been run for awhile.

I suggest we proceed with what we have now to unblock us and fix the tests as a separate task.

Reply:
Looks like we also need to add this for `yarn build` to work.

```
        env:
          NODE_OPTIONS: "--openssl-legacy-provider"
```

I copied it in from webhook lambda.